### PR TITLE
tests: create table from source: platform checks and testdrive

### PR DESCRIPTION
**Pull request by nrainer-materialize**
### Nightly
https://buildkite.com/materialize/nightly/builds?branch=nrainer-materialize%3Atests%2Fcreate-table-from-source

### CODE REVIEW

File: misc/python/materialize/checks/all_checks/source_tables.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+class TableFromSourceBase(Check):
+
+    def _can_run(self, e: Executor) -> bool:
+        return self.base_version >= MzVersion.parse_mz("v0.116.0-dev")
+
+    def generic_setup(self) -> str:
+        return dedent(
+            """
+                $ postgres-execute connection=postgres://mz_system:materialize@${testdrive.materialize-internal-sql-addr}
+                ALTER SYSTEM SET enable_create_table_from_source = true
```
Comment (def-) [inline misc/python/materialize/checks/all_checks/source_tables.py:27]: Can we enable this by default in all of CI?

**Review by def- — APPROVED** [review]
Any other tests planned? I would have added it in parallel-workload too, or are you already working on that?

**Comment by nrainer-materialize** [comment]
> Any other tests planned? I would have added it in parallel-workload too, or are you already working on that?

Absolutely. I am currently about to migrate all existing tests (and additionally keep the original CDC tests).

File: misc/python/materialize/checks/all_checks/source_tables.py (hunk 1 above)
Comment (nrainer-materialize) [reply misc/python/materialize/checks/all_checks/source_tables.py:27]: This is being done in the other PR.

File: misc/python/materialize/checks/all_checks/source_tables.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+                """
+            )
+        )
+
+    def manipulate(self) -> list[Testdrive]:
+        return [
+            Testdrive(dedent(s))
+            for s in [
+                f"""
+                > CREATE TABLE mysql_table_1 FROM SOURCE mysql_source_{self.suffix} (REFERENCE "public_{self.suffix}"."mysql_source_table_1");
```
Comment (rjobanp) [inline misc/python/materialize/checks/all_checks/source_tables.py:166]: we could add a text columns option here if we made one of the table columns an `enum` value or `year` value which are the 2 types that are unsupported natively and we've added text-columns support for: https://materialize.com/docs/sql/create-source/mysql/#supported-types

**Review by rjobanp — APPROVED** [review]
nice!

**Pull request merged**